pyiron/lammps/structure.py

```diff
@@ -406,58 +406,55 @@ def structure_full(self):
         for el in self._structure.get_species_symbols():
             q_dict[el] = float(self.potential.get("set group {} charge".format(el)))
 
-        species_translate_list = list()
-        sorted_species_list = self._structure.get_species_symbols()
-        for el in self._structure.species:
-            ind = np.argwhere(sorted_species_list == el.Abbreviation).flatten()[-1]
-            species_translate_list.append(ind)
+        # species_translate_list = list()
+        # sorted_species_list = self._structure.get_species_symbols()
+        # for el in self._structure.species:
+        #     ind = np.argwhere(sorted_species_list == el.Abbreviation).flatten()[-1]
+        #     species_translate_list.append(ind)
 
-        # analyze structure to get molecule_ids, bonds, angles etc
-        molecule_lst, bonds_lst, angles_lst = [], [], []
+        species_translate_list = [self.potential.get("group {} type".format(el.Abbreviation))
+                                  for el in self.structure.species]
 
-        # species_lst = structure.get_species_objects()
-        # for id_el, el in enumerate(structure.species):
-        #     el.id = id
-        # el_lst = structure.get_chemical_elements()
+        # Drawing bonds only for water molecules
+        molecule_lst, bonds_lst, angles_lst = [], [], []
```

Comment:
If it is specific for water, we should give an error message if any other molecules are set. I guess we had this issue with Markus before and maybe Marcel has had a similar experience. The important part is to make it clear what is supported and what is not supported.

Reply:
Not sure how to do this since we can't identify if an atom belongs to a molecule or not. This info is not available from our atoms class. For example, an atom may be an ion in solution and not a molecule. So if we raise an error message, it might be highly misleading. If more molecule types are to be supported, then they really have to be implemented on a case by case basis, unfortunately. Btw Marcels' system will work now with this fix.

Reply:
I am still confused because you explicitly select O and H indices, so should not we raise error messages for other molecules?

Reply:
Yeah, but that is just to estimate if they are any O-H bonds in the structure. So if there are any O atoms in the structure, the function will check if any H atoms are there in its vicinity. If so, it will draw bonds. The atoms will be written even in the absence of any bonds.